app/services/status_message_service.rb

```diff
@@ -0,0 +1,86 @@
+class StatusMessageService
+  include Rails.application.routes.url_helpers
+
+  attr_reader :status_message
+
+  def initialize(params, user)
+    @user = user
+    normalize_aspect_ids(params)
+    normalize_public_flag!(params)
+    @destination_aspect_ids = destination_aspect_ids(params)
+    @status_message = user.build_post(:status_message, params[:status_message])
+
+    build_location(params)
+    build_poll(params)
+    build_photos(params)
+    build_services(params)
+  end
+
+  def save
+    @status_message.save
+  end
+
+  def process_status
```

Comment:
Hmm I suppose in terms of "Adjective + Noun" it would be `process_message` but I hear a lot of people refer to status messages as just "status" (since just "message" would make it seem like something different) and thus `process_status` sounds better to me.

And yeah I guess it isn't clear by the method name that it will return a status message.